src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx

```diff
@@ -1,25 +1,30 @@
 import React from 'react';
 import {uploadStatuses} from '../Upload.constants';
-import {DisplayActions} from '@jahia/ui-extender';
-import {ButtonRenderer} from '~/utils/getButtonRenderer';
 import {OpenInNew} from '@jahia/moonstone/dist/icons';
+import {Button} from '@jahia/moonstone';
 import PropTypes from 'prop-types';
+import {useSelector} from 'react-redux';
 import {isImageFile} from '../../ContentLayout.utils';
 
 const EditButton = props => {
-    const {status, file, t, path} = props;
+    const {status, file, t, uuid} = props;
 
-    if (isImageFile(file.name)) {
+    const {language} = useSelector(state => ({language: state.language}));
+    const location = (window.location.pathname.split('/').length > 1) ? `/${window.location.pathname.split('/')[1]}` : '/';
+    const url = `${location}/content-editor/${language}/edit/${uuid}`;
+
+    if (isImageFile(file.name) && uuid !== null) {
```

Comment:
Can these two ifs bee merged into one?

Reply:
Personally I would put those differently 
As one is checking the uuid of the node it self while the other is checking if file is an image (different purpose)

Reply:
For me it's more about not having complexity of nested IF, in the end the purpose of the if check is to determine if component should be displayed so I think we would be totally ok merging the two.

Reply:
Ah I see where you are going with that
I can combo those into an additional function that call displayEditButton

Reply:
Updated to 1 if statement 👍